doc/en/example/markers.rst

```diff
@@ -393,7 +393,108 @@ The ``--markers`` option always gives you a list of available markers::
     @pytest.mark.tryfirst: mark a hook implementation function such that the plugin machinery will try to call it first/as early as possible.
     
     @pytest.mark.trylast: mark a hook implementation function such that the plugin machinery will try to call it last/as late as possible.
-    
+
+.. _`creating custom markers with extra arguments`:
+
+Creating custom markers with extra arguments
+--------------------------------------------
+
+Below is the config file that will be used in the next examples::
```

Comment:
Isn't `with_args` main (only?) purpose to allow custom markers to receive *functions* as arguments given that the usual `pytest.mark.foo(f)` syntax cannot know if it is supposed to pass function `f` as argument to the `foo` marker or if it should *decorate* `f` with the `foo` marker?

If that's the case I believe the documentation should focus on that aspect of "how to pass functions to custom markers" and keep it short, without too much details other than explaining why the `pytest.mark.foo(f)` cannot be used for the same purpose. This will make it clear why `with_args` exists and easier for users attempting to pass functions to markers to find it while searching the docs. While I greatly appreciate @xuanluong efforts of explaining things thoroughly here, I feel the docs as they stand are too verbose and don't convey the main purpose of `with_args` prominently.

@xuanluong and @RonnyPfannschmidt what do you guys think?

Reply:
@nicoddemus I see your point. I can make the part discussing the similarity of `pytest.mark.foo` and `pytest.mark.foo.with_args` brief and without the code.

@RonnyPfannschmidt what do you think?

Reply:
Thanks @xuanluong for the understanding. 👍 

@RonnyPfannschmidt what do you think?

Reply:
sounds like it would be more clear than the proposal, please proceed, and thanks 👍